packages/ag-charts-community/src/module/optionsModule.ts

```diff
@@ -100,9 +102,22 @@ export class ChartOptions<T extends AgChartOptions = AgChartOptions> {
         let options = deepClone(userOptions, cloneOptions);
 
         if (this.presetType != null) {
-            const presetOptions = (PRESETS as any)[this.presetType]?.(options, () => this.activeTheme) ?? options;
+            const presetConstructor: ((options: AgPresetOptions, activeTheme: () => ChartTheme) => T) | undefined = (
+                PRESETS as any
+            )[this.presetType];
+            const presetParams = options as any as AgPresetOptions;
+            const presetOptions = presetConstructor?.(presetParams, () => this.activeTheme) ?? options;
             this.debug('>>> AgCharts.createOrUpdate() - applying preset', options, presetOptions);
-            options = presetOptions;
+
+            const activeTheme = getChartTheme(options.theme);
```

Comment:
Nit: This seems to be duplicating work on later lines, could we move the initialisation of `this.activeTheme` to before this `if` block and only do it once?

Reply:
Not quite - because financial charts defines `theme` in its returned options